public/main/group/group_creation.php

```diff
@@ -247,7 +278,7 @@ function copy_value(key) {
     /*
      * Show form to generate new groups
      */
-    $create_groups_form = new FormValidator('create_groups', 'post', api_get_self().'?'.api_get_cidreq());
+    $create_groups_form = new FormValidator('create_groups', 'post', api_get_self().'?'.api_get_cidreq().(isset($_GET['category_id']) ? '&category_id='.$_GET['category_id'] : ''));
```

Comment:
When using $_GET[something], always filter it before using it as part of a URL. Here, because we *know* it's an integer (because it ends with `_id`), you can immediately and very cheaply filter it with `(int) $_GET['category_id']` directly inside the concatenation.